src/openforms/forms/api/serializers/logic/action_serializers.py

```diff
@@ -189,6 +190,19 @@ def validate(self, data: dict) -> dict:
                 code="blank",
             )
 
+        if action_type and action_type == "variable":
+            form_var = FormVariable.objects.get(key=variable)
```

Comment:
This will look through all FormVariables (not only form variables related to this particular form). So if you have another form with a variable with the same key, this will give an error.